src/celeritas/global/alongstep/detail/EnergyLossFluctApplier.hh

```diff
@@ -68,48 +68,56 @@ EnergyLossFluctApplier::operator()(CoreTrackView const& track,
     CELER_EXPECT(step_limit->step > 0);
 
     auto phys = track.make_physics_view();
-    auto ppid = phys.eloss_ppid();
-    if (!ppid)
+    if (!phys.eloss_ppid())
     {
         // No energy loss process for this particle type
         return;
     }
 
+    Energy eloss;
+
     auto particle = track.make_particle_view();
     if (particle.energy() < phys.scalars().eloss_calc_limit)
     {
         // Immediately stop low-energy particles
-        auto step = track.make_physics_step_view();
-        step.deposit_energy(particle.energy());
-        particle.energy(zero_quantity());
-        return;
+        // TODO: this should happen before creating tracks from secondaries
+        // *OR* after slowing down tracks: duplicated in
```

Comment:
Why is this?

Reply:
It seemed to me that applying the cut after the geometry step and before doing the next energy loss step makes it more likely that we'll kill tracks on the boundary, and we're doing less work per step. If we apply the cut at the end of the step (like we originally did with our cutoff params, even later than my comment here suggests), we're more likely to kill because of a discrete interaction or slowing down.

Reply:
Not sure whether I am on the same page, but we should clearly distinguish the tracking cut vs. the production cut (on secondaries from discrete interactions).